### DISCUSSION TITLE
Quick question on timers & unschedule

### DISCUSSION THREAD

**Discussion by GamesStudioTwo** [axmolengine/axmol#1952] [General]
Hi guys I'm just doing timers Node class can I just stick in the unschedule in my class destructor as normal or is the on OnEnd() function I have to use, when I destroy a the Node.

**Comment by rh101** [comment]
I'm not sure what `onEnd()`, or do you mean `onExit()`? The `Node` class already calls the following in the `Node::~Node()` destructor:

```
unscheduleAllCallbacks();
AX_SAFE_RELEASE_NULL(_scheduler);
```

If that is what you mean?

**Reply by GamesStudioTwo** [reply]
Yeah I was just trying to delete the thread, I just found onExit() was what I was looking for,  Thanks
